internal/server/providers.go

```diff
@@ -107,6 +107,20 @@ func (a *API) CreateProvider(c *gin.Context, r *api.CreateProviderRequest) (*api
 	return provider.ToAPI(), nil
 }
 
+func (a *API) PatchProvider(c *gin.Context, r *api.PatchProviderRequest) (*api.Provider, error) {
```

Comment:
We were thinking that put is better, since in the future we want to support single editable fields because it's the most common use case (it's much rarer for people to want to edit 2+ fields) 

EG) so you click on 'provider name' and you can click save

With put, you don't need to know the current state in the backend, you can just send that single field and it'll update that in the backend
With patch, ui will need to pull the data again, just to edit a single field, which also means that it's not as easily supportable to edit a single field at a time 

If there are other use cases for patch, I'm happy to keep it, but for the ui I think put is more useful (at least for now, for  provider)

Reply:
oops, you are correct, I flipped the PUT / PATCH 
this is what I meant: 

```With PATCH, you don't need to know the current state in the backend, you can just send that single field and it'll update that in the backend
With PUT, ui will need to pull the data again, just to edit a single field, which also means that it's not as easily supportable to edit a single field at a time```

Reply:
OK with that in mind, we specifically omitted a PATCH-like operation since it's expected that users will first retrieve the entire resource before making modifications. An exception is user group membership since it's adding/removing user(s) from a list.

I suggest not implement the PATCH endpoint, instead using PUT.

> With PUT, ui will need to pull the data again, just to edit a single field, which also means that it's not as easily supportable to edit a single field at a time

It already needs the provider data to display in the UI so the entire resource should be available for modification. Also why is there a focus on modifying a single field? How is that different than modifying multiple fields?

Reply:
The problem here is that SECRET is not returned from the API. 
So the form will populate with a null value (placeholder ****) --> you are now submitting an updated provider with a blank secret.

This means then now, the PUT api will have to detect if secret field is null, and will not be able to decide if user is a) missing a required field for save to be successful, or b) user does NOT want to update it. 

We can argue that B is the truth, but implementing that functionality (null = do not update) is a patch-behaviour rather than put.